compiler/rustc_middle/src/query/mod.rs

```diff
@@ -2202,6 +2202,11 @@ rustc_queries! {
     query generics_require_sized_self(def_id: DefId) -> bool {
         desc { "check whether the item has a `where Self: Sized` bound" }
     }
+
+    query cross_crate_inlinable(def_id: DefId) -> bool {
+        desc { "whether the item should be made inlinable across crates" }
+        separate_provide_extern
+    }
```

Comment:
I'm not super fond of the name.
Suggestion: `export_optimized_mir`?

Reply:
This query is used to determine 4 things:

* If we codegen the function in all CGUs that reference it
* If we should encode MIR for the function
* If the function is eligible to be inlined in MIR
* If reachabilty analysis should recurse through the function

We do not have name for this collective group of effects or properties. The names `cross_crate_inlinable` and `export_optimized_mir` just call attention to one aspect or another. People looking for better final codegen will care about this query even if MIR inlining is disabled, and people looking for better compile times will care about this query even if LLVM inlining is disabled.

Also, this adds an unstable flag `-Zcross-crate-inline-threshold` which I fully expect people to use at some point (when the cost model here is a bit better) in order to fix missing LLVM inlining. I do not think that calling the flag `-Zexport-optimized-mir-threshold` or having different names for the flag and the query is an improvement.

I would really like to have a name for the whole concept here.